editor/lib/data/particular_editor_controller.dart

```diff
@@ -1,122 +1,55 @@
-      "maxParticles": maxParticles,
-      "sourcePositionVariancex": sourcePositionVarianceX,
-      "sourcePositionVariancey": sourcePositionVarianceY,
-      "startParticleSize": startSize,
-      "startParticleSizeVariance": startSizeVariance,
-      "finishParticleSize": finishSize,
-      "finishParticleSizeVariance": finishSizeVariance,
-      "speed": speed,
-      "speedVariance": speedVariance,
-      "emitterX": emitterX,
-      "emitterY": emitterY,
-      "gravityx": gravityX,
-      "gravityy": gravityY,
-      "minRadius": minRadius,
-      "minRadiusVariance": minRadiusVariance,
-      "maxRadius": maxRadius,
-      "maxRadiusVariance": maxRadiusVariance,
-      "angle": angle,
-      "angleVariance": angleVariance,
-      "rotatePerSecond": rotatePerSecond,
-      "rotatePerSecondVariance": rotatePerSecondVariance,
-      "rotationStart": startRotation,
-      "rotationStartVariance": startRotationVariance,
-      "rotationEnd": finishRotation,
-      "rotationEndVariance": finishRotationVariance,
-      "radialAcceleration": radialAcceleration,
-      "radialAccelVariance": radialAccelerationVariance,
-      "tangentialAcceleration": tangentialAcceleration,
-      "tangentialAccelVariance": tangentialAccelerationVariance,
+  /// The default texture for the particle system.
+  Future<ui.Image> getDefaultTexture() async {
+    if (defaultTexture == null) {
+      ByteData bytes = await rootBundle.load("assets/texture.png");
+      defaultTextureBytes = bytes.buffer.asUint8List();
+      defaultTexture = await loadUIImage(defaultTextureBytes!);
     }
-      ..addAll(startColorMap)
-      ..addAll(startColorVarianceMap)
-      ..addAll(finishColorMap)
-      ..addAll(finishColorVarianceMap);
+    return defaultTexture!;
   }
-}
 
-/// Extension methods for `ARGB` class
-extension ARGBExtension on ARGB {
-  /// Convert to Map for export
-  Map toMap(String name) {
-    return {
-      "${name}Alpha": a,
-      "${name}Red": r,
-      "${name}Green": g,
-      "${name}Blue": b
-    };
+  /// Adds a new particle system to the application.
+  @override
+  Future<void> addConfigs({Map<String, dynamic>? configsData}) async {
+    ByteData bytes;
+
+    /// Load particle texture
+    ui.Image? texture;
+    try {
+      if (configsData != null && configsData.containsKey("textureFileName")) {
+        bytes =
+            await rootBundle.load("assets/${configsData["textureFileName"]}");
+        texture = await loadUIImage(bytes.buffer.asUint8List());
+      }
+    } catch (e) {
+      debugPrint(e.toString());
+    }
+    final configs = ParticularConfigs()..initialize(configs: configsData);
+
+    final layer = ParticularEditorLayer(
+      texture: texture ?? await getDefaultTexture(),
+      textureBytes: defaultTextureBytes,
+      configs: configs,
+    );
+
+    if (configsData == null || !configsData.containsKey("configName")) {
+      configs.updateFromMap({"configName": "Layer ${layers.length + 1}"});
```

Comment:
can you change updateFromMap to copyWith and maybe with factory method

Reply:
This method is not copy or clone method. I change to `updateWith`. but if is there better example I really appreciated from.